src/main/webapp/app/components/tabs/curation-data-import-tab/import-methods.tsx

```diff
@@ -0,0 +1,329 @@
+): Promise<DataImportStatus> => {
+  // validate duplication
+  const hugoSymbol = dataRow.data.hugo_symbol;
+  const mutation = new Mutation(dataRow.data.alteration);
+  let mutationImpactStatusUpdated = false;
+
+  const request: AnnotateAlterationBody[] = [
+    {
+      referenceGenome: REFERENCE_GENOME.GRCH37,
+      alteration: { alteration: mutation.name, genes: [{ hugoSymbol } as Gene] } as ApiAlteration,
+    },
+  ];
+  try {
+    const annotatedAlterations = await flowResult(flow(alterationStore.annotateAlterations)(request));
+    const annotatedAlteration: Alteration = annotatedAlterations[0].entity;
+    mutation.alterations = [
+      {
+        type: annotatedAlteration.type ?? AlterationTypeEnum.Unknown,
+        alteration: dataRow.data.alteration,
+        name: dataRow.data.name ?? annotatedAlteration.name,
+        consequence: annotatedAlteration.consequence?.name ?? '',
+        comment: '',
+        excluding: [],
+        genes: annotatedAlteration.genes || [],
+        proteinChange: dataRow.data.protein_change ?? annotatedAlteration.proteinChange,
+        proteinStart: annotatedAlteration.start,
+        proteinEnd: annotatedAlteration.end,
+        refResidues: annotatedAlteration.refResidues || '',
+        varResidues: annotatedAlteration.variantResidues || '',
+      },
+    ];
+  } catch (e) {
+    return {
+      status: 'error',
+      message: "We aren't able to annotate the alteration. Please reach out to dev team.",
+    };
+  }
+
+  const isGermlineData = (data: GermlineMutationDI | SomaticMutationDI): data is GermlineMutationDI => {
+    return isGermline;
+  };
+  if (isGermlineData(dataRow.data)) {
+    const data = dataRow.data;
+    if (data.pathogenicity) {
+      const importStatus: DataImportStatus = pathogenicityCheck(data.pathogenicity);
+      if (importStatus.status === 'error') {
+        return importStatus;
+      }
+    }
+    mutation.mutation_effect.pathogenic = data.pathogenicity as PATHOGENICITY;
+    mutation.mutation_effect.pathogenic_review = new Review(authStore.fullName, '');
+    mutationImpactStatusUpdated = true;
+  } else {
+    const data = dataRow.data;
+    if (data.oncogenicity) {
+      const importStatus: DataImportStatus = oncogenicityCheck(data.oncogenicity);
+      if (importStatus.status === 'error') {
+        return importStatus;
+      }
+    }
+    mutation.mutation_effect.oncogenic = data.oncogenicity as FIREBASE_ONCOGENICITY;
+    mutation.mutation_effect.oncogenic_review = new Review(authStore.fullName, '');
+    mutationImpactStatusUpdated = true;
+  }
+
+  const existingMuts = getDuplicateMutations([mutation.name], mutationList, vusList, {
+    useFullAlterationName: true,
+    exact: true,
+  });
+
+  // Delete existing VUS before importing
+  for (const mut of existingMuts) {
+    if (mut.inVusList) {
+      await firebaseGeneService.firebaseRepository.delete(`${getFirebaseVusPath(isGermline, hugoSymbol)}/${mut.duplicate}`);
+    } else {
```

Comment:
I'm wondering if instead of deleting the VUS immediately, we mark the VUS as `promotedToMutation: true` in Review object, so that it can be reviewed?